Object.assing(this, {...}) not interpreted in constructors

For this file:
```javascript
class T {
    a: string
    constructor (a:string) {
	Object.assign(this, {a})
    }
}

let t = new T('b')
```
tsc 3.8.3 reports

`error TS2564: Property 'a' has no initializer and is not definitely assigned in the constructor.`

I think property 'a'is definitely assigned in the constructor. `this.a = a` means the same.

--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -3061,7 +3061,7 @@ namespace ts {
         if (!(name.kind === SyntaxKind.ComputedPropertyName || name.kind === SyntaxKind.ElementAccessExpression)) {
             return false;
         }
-        const expr = isElementAccessExpression(name) ? name.argumentExpression : name.expression;
+        const expr = isElementAccessExpression(name) ? skipParentheses(name.argumentExpression) : name.expression;
         return !isStringOrNumericLiteralLike(expr) &&
             !isSignedNumericLiteral(expr) &&
             !isWellKnownSymbolSyntactically(expr);

--- a/tests/baselines/reference/propertyAssignmentOnParenthesizedNumber.symbols
+++ b/tests/baselines/reference/propertyAssignmentOnParenthesizedNumber.symbols
@@ -1,0 +1,8 @@
+=== tests/cases/conformance/salsa/bug38934.js ===
+var x = {};
+>x : Symbol(x, Decl(bug38934.js, 0, 3))
+
+// should not crash and also should not result in a property '0' on x.
+x[(0)] = 1;
+>x : Symbol(x, Decl(bug38934.js, 0, 3))
+

--- a/tests/baselines/reference/propertyAssignmentOnParenthesizedNumber.types
+++ b/tests/baselines/reference/propertyAssignmentOnParenthesizedNumber.types
@@ -1,0 +1,14 @@
+=== tests/cases/conformance/salsa/bug38934.js ===
+var x = {};
+>x : {}
+>{} : {}
+
+// should not crash and also should not result in a property '0' on x.
+x[(0)] = 1;
+>x[(0)] = 1 : 1
+>x[(0)] : any
+>x : {}
+>(0) : 0
+>0 : 0
+>1 : 1
+

--- a/tests/cases/conformance/salsa/propertyAssignmentOnParenthesizedNumber.ts
+++ b/tests/cases/conformance/salsa/propertyAssignmentOnParenthesizedNumber.ts
@@ -1,0 +1,8 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+// @Filename: bug38934.js
+
+var x = {};
+// should not crash and also should not result in a property '0' on x.
+x[(0)] = 1;

